src/rt/region/region_rc.h

```diff
@@ -467,33 +600,39 @@ namespace verona::rt
       ObjectStack dfs(alloc);
       ObjectStack sub_regions(alloc);
       o->trace(dfs);
-      ObjectStack fin_q(alloc);
+      LinkedObjectStack gc;
 
-      RegionRc* reg = get(in);
+      RegionRc* reg = (RegionRc*)opened_region();
 
-      fin_q.push(o);
+      gc.push(o);
 
       while (!dfs.empty())
       {
         Object* p = dfs.pop();
         switch (p->get_class())
         {
-          case Object::ISO:
-            if (p == in)
-            {
-              reg->entry_point_count->metadata -= 1;
-            }
-            else
+          case Object::OPEN_ISO:
+            if (decref_inner(p))
             {
-              sub_regions.push(p);
+              // We have to halt the deallocation and instead start a full
```

Comment:
This is an interesting question. Is it possible to trigger this case in the wild? I couldn't think of a reason why you'd decref an ISO to 0 since it's the entry point and you'd call release for that. _If_ we do want to support it, then it's important to delegate this to `release_internal` because that also cleans up cyclic garbage.